contrib/epee/include/storages/parserse_base_utils.h

```diff
@@ -157,7 +157,6 @@ namespace misc_utils
         while (fi != buf_end && ((lut[(uint8_t)*fi] & 32)) == 0)
           ++fi;
         val.assign(it, fi);
-        val.reserve(std::distance(star_end_string, buf_end));
```

Comment:
Was this actually slower ?

Reply:
This was slower because typically the json string doesn't have any escapes so this usually creates an unnecessary allocation. This could arguably be moved above the assign, but then the json parser is typically requesting enough memory for the remainder of the json buffer which is a big over estimate (and seems to be harder on heap implementations). The `push_back` call will typically double the space needed each time the capacity is reached so the performance drop on escape handling is lessened.

Without this change, moving the json string values is actually slower due to the time spent searching within the heap (especially when the incoming json buffer is large). The difference is that the large allocation is not free'd after a copy (it is now moved and kept in use), and therefore the heap has to find another large chunk of memory instead of re-using a recently used large chunk. I wouldn't rely on this behavior because its fairly specific to the glibc I happen to be running during the tests - it is very repeatable on that box though. And it was a b**** to track down because reducing allocations + copies pretty much never increases wall clock time (except for when it does because heaps are a PITA).

Reply:
Ah, I might have tested with worst case inputs :)